# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.h
# Commit: e06fd5d60970
# Full Hash: e06fd5d60970eb7dcda55192c778ac2230c2b201
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-30 04:06:43
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Implement VideoDecoder::Close r=padenot
#   
#   This patch implements the close method [1] for WebCodecs' VideoDecoder
#   interface. The implementation has not been completed yet. It will be
#   done by the following patches.
# ==============================================================================

diff -r 58a0e543998b -r e06fd5d60970 dom/media/webcodecs/VideoDecoder.h
--- a/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 22:33:11 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 22:33:11 2023 +0000
@@ -73,7 +73,7 @@
 
   void Reset(ErrorResult& aRv);
 
-  void Close(ErrorResult& aRv);
+  MOZ_CAN_RUN_SCRIPT void Close(ErrorResult& aRv);
 
   static already_AddRefed<Promise> IsConfigSupported(
       const GlobalObject& aGlobal, const VideoDecoderConfig& aConfig,
@@ -84,6 +84,9 @@
   void AssertIsOnOwningThread() const { NS_ASSERT_OWNINGTHREAD(VideoDecoder); }
 
   Result<Ok, nsresult> Reset(const nsresult& aResult);
+  MOZ_CAN_RUN_SCRIPT Result<Ok, nsresult> Close(const nsresult& aResult);
+
+  MOZ_CAN_RUN_SCRIPT void ReportError(const nsresult& aResult);
 
   // Constant in practice, only set in ::Constructor.
   RefPtr<WebCodecsErrorCallback> mErrorCallback;
